python/langsmith/_expect.py

```diff
@@ -0,0 +1,359 @@
+            "to_be_less_than",
+        )
+
+    def to_be_greater_than(self, value: float) -> None:
+        """Assert that the expectation value is greater than the given value.
+
+        Args:
+            value: The value to compare against.
+
+        Raises:
+            AssertionError: If the expectation value is not
+            greater than the given value.
+        """
+        self._assert(
+            self.value > value,
+            f"Expected {self.key} to be greater than {value}, but got {self.value}",
+            "to_be_greater_than",
+        )
+
+    def to_be_between(self, min_value: float, max_value: float) -> None:
+        """Assert that the expectation value is between the given min and max values.
+
+        Args:
+            min_value: The minimum value (exclusive).
+            max_value: The maximum value (exclusive).
+
+        Raises:
+            AssertionError: If the expectation value
+                is not between the given min and max.
+        """
+        self._assert(
+            min_value < self.value < max_value,
+            f"Expected {self.key} to be between {min_value} and {max_value},"
+            f" but got {self.value}",
+            "to_be_between",
+        )
+
+    def to_be_approximately(self, value: float, precision: int = 2) -> None:
+        """Assert that the expectation value is approximately equal to the given value.
+
+        Args:
+            value: The value to compare against.
+            precision: The number of decimal places to round to for comparison.
+
+        Raises:
+            AssertionError: If the rounded expectation value
+                does not equal the rounded given value.
+        """
+        self._assert(
+            round(self.value, precision) == round(value, precision),
+            f"Expected {self.key} to be approximately {value}, but got {self.value}",
+            "to_be_approximately",
+        )
+
+    def to_equal(self, value: float) -> None:
+        """Assert that the expectation value equals the given value.
+
+        Args:
+            value: The value to compare against.
+
+        Raises:
+            AssertionError: If the expectation value does
+                not exactly equal the given value.
+        """
+        self._assert(
+            self.value == value,
+            f"Expected {self.key} to be equal to {value}, but got {self.value}",
+            "to_equal",
+        )
+
+    def to_contain(self, value: Any) -> None:
+        """Assert that the expectation value contains the given value.
+
+        Args:
+            value: The value to check for containment.
+
+        Raises:
+            AssertionError: If the expectation value does not contain the given value.
+        """
+        self._assert(
+            value in self.value,
+            f"Expected {self.key} to contain {value}, but it does not",
+            "to_contain",
+        )
+
+    # Custom assertions
+    def against(self, func: Callable, /) -> None:
```

Comment:
@nfcampos I'm OK dropping or renaming  this one if it's premature. Basically wanted an easy way to log feedback in a unit test context